lib/charms/opensearch/v0/opensearch_distro.py

```diff
@@ -430,3 +430,7 @@ def missing_sys_requirements() -> List[str]:
             missing_requirements.append("net.ipv4.tcp_retries2 should be 5")
 
         return missing_requirements
+
+    def version(self) -> str:
+        """Returns the version number of this opensearch instance."""
+        return self.request("GET", "/").get("version").get("number")
```

Comment:
I think you need to handle an `OpenSearchHttpError` here.

I wonder, if we should not actually store the version and the static stuff in the unit peer data bag? basically at the end of node initialization, we could put the version on the unit data. So that we only make an HTTP request to fetch the version. What do you think?

Reply:
You're right on the HttpError, I'll add that in.

This is actually stored in the client relation databag when the relation is created, but if we want to reduce requests to opensearch I'm happy to store it in the peer databag as well.

Reply:
Hmm, then I think for the time being the relation databag is enough, until we need the version on a broader context (opensearch version upgrades etc..)

Reply:
**suggestion**: could this be a property? actually, following this discussion, another option could also be to use the cached_property [decorator](https://docs.python.org/3/library/functools.html#functools.cached_property), where you may optimize things a bit

Reply:
Since the charm object is rebuilt every time a hook fires, I don't know how useful the caching in cached_property will be, especially if this function is only called when we set up the client relations so far. Caching this across hooks means something like storing it in StoredState or a unit databag, but this fires infrequently enough that I don't feel it's necessary until we're adding version upgrades. 

I'm probably going to set this to be a normal property - TWIL about `cached_property` though!